src/pytest_bdd/parser.py

```diff
@@ -337,25 +339,27 @@ class Examples:
 
     line_number: int | None = field(default=None)
     name: str | None = field(default=None)
+    tags: set[str] = field(default_factory=set)
 
     example_params: list[str] = field(init=False, default_factory=list)
+    example_tags: list[str] = field(init=False, default_factory=list)
     examples: list[Sequence[str]] = field(init=False, default_factory=list)
 
     def set_param_names(self, keys: Iterable[str]) -> None:
         self.example_params = [str(key) for key in keys]
 
     def add_example(self, values: Sequence[str]) -> None:
         self.examples.append(values)
+        self.example_tags.append(self.tags)
```

Comment:
Seems like this would add the tags multiple times? Or is the idea that the "examples" and "example_tags" arrays are parallel arrays, related by index?

If the latter, seems like it might be better to modify the data structure a bit more, such that we have each set of examples as its own object, which as its own set of associated tags.

Reply:
Hmmm. Actually, now that I look at the entirety of the code, I think I can see why I did it this way. And, if I'm understanding everything correctly, I think I agree with you. :-D